oidc_setup.sh

```diff
@@ -0,0 +1,571 @@
+
+set -e
+
+INTEGRATION_SETUP_DEFAULT_TEMP_DIR=./temp
+
+# helper functions
+help() {
+  echo -e "Available environment variables:"
+  echo -e "Nextcloud:"
+  echo -e "\t NEXTCLOUD_HOST \t\t\t Nextcloud host URL"
+  echo -e "\t NC_ADMIN_USERNAME \t\t\t Nextcloud admin username"
+  echo -e "\t NC_ADMIN_PASSWORD \t\t\t Nextcloud admin password"
+  echo -e "\t NC_INTEGRATION_PROVIDER_NAME \t\t SSO Provider name (Not required when using 'nextcloud_hub' type)"
+  echo -e "\t NC_INTEGRATION_PROVIDER_TYPE \t\t Single Sign-On provider type ('nextcloud_hub' or 'external')"
+  echo -e "\t NC_INTEGRATION_OP_CLIENT_ID \t\t OpenProject client ID (Not required when using 'nextcloud_hub' type)"
+  echo -e "\t NC_INTEGRATION_TOKEN_EXCHANGE \t\t Enable token exchange (true/false) (Not required when using 'nextcloud_hub' type)"
+  echo -e "\t SETUP_PROJECT_FOLDER \t Enable project folder (true/false)"
+  echo -e ""
+  echo -e "OpenProject:"
+  echo -e "\t OPENPROJECT_HOST \t OpenProject host URL"
+  echo -e "\t OP_ADMIN_USERNAME \t OpenProject admin username"
+  echo -e "\t OP_ADMIN_PASSWORD \t OpenProject admin password"
+  echo -e "\t OPENPROJECT_STORAGE_NAME \t OpenProject file storage name (eg: nextcloud)"
+}
+
+log_error() {
+  echo -e "\e[31m$1\e[0m"
+}
+
+log_info() {
+  echo -e "\e[37m$1\e[0m"
+}
+
+log_success() {
+  echo -e "\e[32m$1\e[0m"
+}
+
+if [[ -z "$NEXTCLOUD_HOST" || -z "$OPENPROJECT_HOST" ]]; then
+  log_error "Nextcloud and OpenProject host URLs are required."
+  help
+  exit 1
+fi
+
+if [[ -z "$NC_ADMIN_USERNAME" || -z "$NC_ADMIN_PASSWORD" ]]; then
+  log_error "Nextcloud admin username and password are required."
+  help
+  exit 1
+fi
+
+# Validate required configs for integration setup
+if [[ -z $NC_INTEGRATION_PROVIDER_TYPE ]] ||
+  [[ -z $SETUP_PROJECT_FOLDER ]] ; then
+  log_error "Following configs are required for integration setup:"
+  log_error "\tSETUP_PROJECT_FOLDER"
+  log_error "\tNC_INTEGRATION_PROVIDER_TYPE"
+  help
+  exit 1
+fi
+
+# Support for "Debug mode"
+if [[ $INTEGRATION_SETUP_DEBUG == "true"  ]]; then
+  log_info "Debug mode is enabled"
+  set -x
+  set -v
+fi
+
+if [[ $INTEGRATION_SETUP_TEMP_DIR == ""  ]]; then
+  log_info "Using default temp dir: ${INTEGRATION_SETUP_DEFAULT_TEMP_DIR}"
+  INTEGRATION_SETUP_TEMP_DIR=${INTEGRATION_SETUP_DEFAULT_TEMP_DIR}
+  mkdir -p ${INTEGRATION_SETUP_TEMP_DIR}
+else
+  log_info "Using ${INTEGRATION_SETUP_TEMP_DIR} as non-default temporary directory"
+  if [ ! -d "${INTEGRATION_SETUP_DEFAULT_TEMP_DIR}" ]; then
+    log_error "Temporary directory does not exist"
+    exit 1
+  fi
+fi
+
+# if something goes wrong or an error occurs during the setup of the whole integration
+# we can delete the storage created in OpenProject, otherwise it would need to be deleted manually when the script is run the next time
+deleteOPStorageAndPrintErrorResponse() {
```

Comment:
prefix the function name as well
```suggestion
opDeleteStorage() {
```

Reply:
The comment is meant for other functions as well. try to namespace the variables and function names